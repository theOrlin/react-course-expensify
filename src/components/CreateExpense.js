import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { addExpense } from './../actions/expenses';

export class CreateExpense extends React.Component {
    render() {
        return (
            <div>
                <h1>Add Expense</h1>
                <ExpenseForm
                    onSubmit={this.onSubmit}
                />
            </div>
        )
    }
    onSubmit = (expense) => {
        // props.dispatch(addExpense(expense));
        this.props.addExpense(expense);
        this.props.history.push('/');
    }
}

const mapDispatchToProps = (dispatch) => ({
    addExpense: (expense) => dispatch(addExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(CreateExpense);