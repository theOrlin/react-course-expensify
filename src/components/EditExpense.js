import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from './../actions/expenses';

export class EditExpense extends React.Component {
    render() {
        return (
            <div>
                <ExpenseForm
                    expense={this.props.expense}
                    onSubmit={this.onSubmit} />
                <button onClick={(e) => {
                    this.onRemove(this.props.expense.id);
                }}>Remove</button>
            </div>
        );
    };

    onSubmit = (expense) => {
        this.props.editExpense(this.props.expense.id, expense);
        this.props.history.push('/');
    };

    onRemove = (id) => {
        this.props.removeExpense(id);
        this.props.history.push('/');
    }
};

const mapDispatchToProps = (dispatch) => {
    return {
        editExpense: (id, expense) => dispatch(editExpense(id, expense)),
        removeExpense: (id) => dispatch(removeExpense({ id }))
    }
};

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find(expense => expense.id === props.match.params.id)
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpense);