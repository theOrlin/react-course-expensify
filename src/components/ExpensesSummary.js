import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpencesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';

export const ExpensesSummary = (props) => (
    <p>
    Viewing {props.expenses.length} expense{props.expenses.length > 1 ? 's' : ''} totalling {numeral(selectExpencesTotal(props.expenses) / 100).format('0,0.00')}</p>
)

const mapStateToProps = (state) => ({
    expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpensesSummary);