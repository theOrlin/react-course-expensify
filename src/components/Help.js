import React from 'react';

const Help = () => (
    <div>This is from Help</div>
)

export default Help;