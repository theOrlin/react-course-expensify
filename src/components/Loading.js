import React from 'react';

const Loading = () => (
    <div className="loader">
        <img className="loader__image" src="/images/loader.gif" alt="loading spinner"/>
    </div>
);

export default Loading;