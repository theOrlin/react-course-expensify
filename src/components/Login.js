import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const Login = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>Get your expenses under control</p>
            <button onClick={startLogin} className="button">Log In with Google</button>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
})

export default connect(undefined, mapDispatchToProps)(Login);