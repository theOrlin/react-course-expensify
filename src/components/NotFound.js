import React from 'react';
import { Link } from 'react-router-dom';

const NotFound = () => (
    <div>
        <h1>404!</h1>
        <div>
            <Link to="/">Go home</Link>
        </div>
    </div>
)

export default NotFound;
