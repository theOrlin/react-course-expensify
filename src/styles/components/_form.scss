.form {
    display: flex;
    flex-direction: column;

    >* {
        margin-bottom: $m-size;
    }

    .SingleDatePickerInput {
        display: block;
    }

    .DateInput.DateInput_1 {
        width: auto;
        display: block;
    }
}

.form__error {
    font-style: italic;
    margin: 0 0 $m-size 0;
}