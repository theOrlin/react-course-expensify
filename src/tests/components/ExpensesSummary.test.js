import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should render proper message for a single expense', () => {
    const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render proper message for multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});
