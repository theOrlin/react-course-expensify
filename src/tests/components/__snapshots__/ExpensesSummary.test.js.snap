// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render proper message for a single expense 1`] = `
<p>
  Viewing 
  1
   expense
   totalling 
  1.95
</p>
`;

exports[`should render proper message for multiple expenses 1`] = `
<p>
  Viewing 
  3
   expense
  s
   totalling 
  1,141.95
</p>
`;
